src/systems/sound-effects-system.js

```diff
@@ -0,0 +1,245 @@
+        held: null
+      }
+    };
+    this.teleporterState = {
+      leftHand: {
+        teleporting: false,
+        teleportArcSource: null
+      },
+      rightHand: {
+        teleporting: false,
+        teleportArcSource: null
+      },
+      rightRemote: {
+        teleporting: false,
+        teleportArcSource: null
+      }
+    };
+    this.ctx = THREE.AudioContext.getContext();
+    this.pendingEffects = [];
+    this.soundFor = new Map();
+    this.sounds = {};
+    getBuffer(TICK, this.ctx).then(buffer => {
+      this.sounds.tick = buffer;
+      this.soundFor.set("thaw", buffer);
+      this.soundFor.set("pen_stop_draw", buffer);
+      this.soundFor.set("pen_undo_draw", buffer);
+      this.soundFor.set("pen_stop_draw", buffer);
+      this.soundFor.set("pen_change_radius", buffer);
+      this.soundFor.set("pen_change_color", buffer);
+      this.soundFor.set("toggle_mic", buffer);
+    });
+    getBuffer(TELEPORT_ARC, this.ctx).then(buffer => {
+      this.sounds.teleportArc = buffer;
+    });
+    getBuffer(QUICK_TURN, this.ctx).then(buffer => {
+      this.sounds.teleportEnd = buffer;
+    });
+    getBuffer(TAP_MELLOW, this.ctx).then(buffer => {
+      this.sounds.snapRotate = buffer;
+      this.soundFor.set("snap_rotate_left", buffer);
+      this.soundFor.set("snap_rotate_right", buffer);
+    });
+    getBuffer(PEN_SPAWN, this.ctx).then(buffer => {
+      this.sounds.spawnPen = buffer;
+      this.soundFor.set("spawn_pen", buffer);
+    });
+    getBuffer(PEN_DRAW, this.ctx).then(buffer => {
+      this.sounds.penStartDraw = buffer;
+      this.soundFor.set("pen_start_draw", buffer);
+    });
+    getBuffer(CAMERA_SNAPSHOT, this.ctx).then(buffer => {
+      this.sounds.cameraSnapshot = buffer;
+      this.soundFor.set("camera_tool_took_snapshot", buffer);
+    });
+    getBuffer(WELCOME, this.ctx).then(buffer => {
+      this.sounds.enterScene = buffer;
+      this.soundFor.set("enter_scene", buffer);
+    });
+    getBuffer(QUACK, this.ctx).then(buffer => {
+      this.sounds.quack = buffer;
+      this.soundFor.set("quack", buffer);
+    });
+    getBuffer(SPECIAL_QUACK, this.ctx).then(buffer => {
+      this.sounds.specialQuack = buffer;
+      this.soundFor.set("special_quack", buffer);
+    });
+    getBuffer(POP, this.ctx).then(buffer => {
+      this.sounds.chatMessage = buffer;
+      this.soundFor.set("chat_message", buffer);
+    });
+    getBuffer(FREEZE, this.ctx).then(buffer => {
+      this.sounds.freeze = buffer;
+      this.soundFor.set("freeze", buffer);
+    });
+    getBuffer(TACK, this.ctx).then(buffer => {
+      this.sounds.pin = buffer;
+      this.soundFor.set("pin", buffer);
+    });
+    getBuffer(MEDIA_LOADING, this.ctx).then(buffer => {
+      this.sounds.mediaLoading = buffer;
+    });
+    getBuffer(MEDIA_LOADED, this.ctx).then(buffer => {
+      this.sounds.mediaLoaded = buffer;
+      this.soundFor.set("media_loaded", buffer);
+    });
+
+    this.mediaLoadingSources = [];
```

Comment:
Hmm surprised these ones are not spatialized sounds. In any case would be nice to figure out a way to handle these sort of looping sounds in the `pendingEffects` queue, but may not be worth bothering with until we have some more examples.